src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/model/canonical/CanonicalRecord.kt

```diff
@@ -45,23 +46,29 @@ data class CanonicalRecord(
 ) {
   companion object {
     fun from(personKey: PersonKeyEntity): CanonicalRecord {
-      val latestPerson = personKey.personEntities.sortedByDescending { it.lastModified }.first()
-      return CanonicalRecord(
-        cprUUID = personKey.personId.toString(),
-        firstName = latestPerson.firstName,
-        middleNames = latestPerson.middleNames,
-        lastName = latestPerson.lastName,
-        dateOfBirth = latestPerson.dateOfBirth?.toString() ?: "",
-        title = latestPerson.title,
-        sex = latestPerson.sex,
-        religion = latestPerson.religion,
-        ethnicity = latestPerson.ethnicity,
-        aliases = CanonicalAlias.fromPseudonymEntityList(latestPerson.pseudonyms),
-        addresses = CanonicalAddress.fromAddressEntityList(latestPerson.addresses),
-        identifiers = getCanonicalIdentifiers(personKey.personEntities) + CanonicalIdentifier.fromReferenceEntityList(latestPerson.references),
-        nationalities = CanonicalNationality.from(latestPerson),
+      val personList = personKey.personEntities.sortedByDescending { it.lastModified }
 
-      )
+      when (personList.isNullOrEmpty()) {
```

Comment:
I think there might be a better place for this logic

https://github.com/ministryofjustice/hmpps-person-record/blob/a603699a9e2619c1865b401a23eef46fe39fc0a5/src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt#L118

we are already checking if the entity exists here (incorrectly BTW - please also fix the check!) so we could just go one step further and check the list of people is not empty too?

Something like this should do the trick?

`  private fun buildCanonicalRecord(personKeyEntity: PersonKeyEntity?, uuid: UUID): CanonicalRecord = when {
    personKeyEntity?.personEntities?.isNotEmpty() == true -> CanonicalRecord.from(personKeyEntity)
    else -> throw PersonKeyNotFoundException(uuid)
  }`

Reply:
Incidentally I have tested the API in dev and this functionality already works as expected